tests/tests.js

```diff
@@ -841,10 +841,28 @@ const namedGroupFixtures = [
 		'expectedGroups': [
 			['name', 1]
 		]
+	},
+	{
+		'pattern': '(?:(?<a>x)|(?<a>y))\\k<a>',
+		'flags': '',
+		'expected': '(?:(x)|(y))\\1\\2',
```

Comment:
@bakkot Is it correct to replace `\k<a>` with `\1\2`?

Reply:
In this case, yes - only the `1` or `2` capturing groups can actually participate in the match. If the `1` group didn't participate, then `\1` will unconditionally match without consuming input; similarly for `\2`. So `\1\2` will always match whichever group actually did participate.